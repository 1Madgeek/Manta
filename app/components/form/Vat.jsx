// Libraries
import React, {Component} from 'react';

// Custom Components
import Switch from '../shared/Switch';
import {Section} from '../shared/Section';

// Animation
import _withFadeInAnimation from '../shared/hoc/_withFadeInAnimation';

// Styles
import styled from 'styled-components';
const VatWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

const VatContent = styled.div`
  display: flex;
  flex-direction: column;
`;

const VatAmount = styled.div`flex: 1;`;

const VatType = styled.div`
  flex: 1;
  margin-top: 10px;
`;

// Component
class Vat extends Component {
  componentWillMount = () => {
    const {vat} = this.props;
    this.setState({
      amount: vat.amount ? vat.amount : '',
    });
  };

  handleInputChange = event => {
    const name = event.target.name;
    const value =
      event.target.value === '' ? '' : parseInt(event.target.value, 10);
    this.setState({[name]: value}, () => {
      this.updateVatState();
    });
  };

  updateVatState = () => {
    const {updateFieldData} = this.props;
    updateFieldData('vat', this.state);
  };

  render = () => {
    return (
      <Section>
        <VatWrapper>
          <label className="itemLabel">Vat (%)</label>
          <VatContent>
            <VatAmount>
              <input
                name="amount"
                type="number"
                value={this.state.amount}
                onChange={e => this.handleInputChange(e)}
                placeholder="Amount"
              />
            </VatAmount>
          </VatContent>
        </VatWrapper>
      </Section>
    );
  };
}

// Exports
export default _withFadeInAnimation(Vat);
