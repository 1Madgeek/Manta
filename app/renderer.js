// Node Libs
const path = require('path');
const glob = require('glob');

// Custom Libs
const sounds = require(path.join(__dirname, '../libs/sounds.js'));

// Imports Main Renderere Files
function loadMaiRendererFiles() {
  const files = glob.sync(path.join(__dirname, './renderers/*.js'));
  files.forEach(file => require(file));
  sounds.preload();
}

function initialize() {
  loadMaiRendererFiles();
  sounds.play('STARTUP');
  console.timeEnd('init');
}

initialize();
