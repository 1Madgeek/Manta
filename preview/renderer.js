// Electron Libs
const ipc = require('electron').ipcRenderer;

// Libraries
import React from 'react';
import ReactDOM from 'react-dom';

// Custom Components
import Receipt from './containers/Receipt.jsx';

// Render Preview Window
ipc.on('update-preview', (event, receiptData) => {
  ReactDOM.render(
    <Receipt receiptData={receiptData} />,
    document.getElementById('receiptWrapper')
  );
});

