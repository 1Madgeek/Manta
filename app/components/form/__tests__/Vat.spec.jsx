// Libs
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

// Component
import {Vat} from '../Vat.jsx';

// Mocks
const updateFieldData = jest.fn();
const vat = {
  required: true,
  amount: 10
};

describe('Note component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Vat vat={vat} updateFieldData={updateFieldData} />);
  });

  // PROPS & STATE
  it('receives correct props', () => {
    expect(wrapper.prop('vat')).toEqual(vat);
    expect(wrapper.prop('updateFieldData')).toEqual(updateFieldData);
  });

  // RENDER
  it('renders necessary element', () => {
    expect(wrapper.find('label')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(1);
  });

  // LIFE CYCLE EVENTS
  // TODO
  it('render when necessary', () => {});

  // PRIVATE METHOD
  it('handleInputChange correctly', () => {
    const spy = jest.spyOn(Vat.prototype, 'handleInputChange');
    const wrap = mount(<Vat vat={vat} updateFieldData={updateFieldData} />);
    const input = wrap.find('input');
    input.simulate('change', { target: { value: '20' } });
    expect(spy).toHaveBeenCalled();
  });

  // SNAPSHOT
  it('matches snapshot', () => {
    const tree = renderer
      .create(<Vat vat={vat} updateFieldData={updateFieldData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
