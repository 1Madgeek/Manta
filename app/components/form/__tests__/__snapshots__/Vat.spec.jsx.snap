// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component matches snapshot 1`] = `
<div
  className="Section__SectionStyle-KuFAH hVJJBo"
>
  <div
    className="Vat__VatWrapper-dhkKOK mcQjd"
  >
    <label
      className="itemLabel"
    >
      Vat (%)
    </label>
    <div
      className="Vat__VatContent-cjUcMd hODULM"
    >
      <div
        className="Vat__VatAmount-fkgylN fyheuP"
      >
        <input
          name="amount"
          onChange={[Function]}
          placeholder="Amount"
          type="number"
          value={10}
        />
      </div>
    </div>
  </div>
</div>
`;
